### PR TITLE
CP-371: Update Kafka version to 0.10.1.0-SNAPSHOT and fix up build issues.

### PR DESCRIPTION
**Pull request by ewencp**
A broker state, RunningAsController, was removed entirely. It is safe to simply remove it from the one condition is was used in.

### CODE REVIEW

**Comment by ewencp** [comment]
@arrawatia Care to review? Should be a quick one.

**Comment by ijuma** [comment]
LGTM

File: support-metrics-common/src/main/java/io/confluent/support/metrics/common/kafka/KafkaUtilities.java [hunk 1]
```diff
@@ -265,8 +264,7 @@ public VerifyTopicState verifySupportTopic(ZkUtils zkUtils,
   }
 
   public boolean isReadyForMetricsCollection(KafkaServer server) {
-    return server.brokerState().currentState() == RunningAsBroker.state() ||
-        server.brokerState().currentState() == RunningAsController.state();
+    return server.brokerState().currentState() == RunningAsBroker.state();
```
Comment (arrawatia) [inline support-metrics-common/src/main/java/io/confluent/support/metrics/common/kafka/KafkaUtilities.java]: Does this mean you are able to create topics and write to it ? Do you not need controller leader election to be over before the cluster is ready to accept client requests?

Comment (ewencp) [inline support-metrics-common/src/main/java/io/confluent/support/metrics/common/kafka/KafkaUtilities.java]: The state was removed in trunk. The only active state is now `RunningAsBroker`.

Comment (arrawatia) [inline support-metrics-common/src/main/java/io/confluent/support/metrics/common/kafka/KafkaUtilities.java]: OK. LGTM then.

Comment (ijuma) [inline support-metrics-common/src/main/java/io/confluent/support/metrics/common/kafka/KafkaUtilities.java]: The `RunningAsController` state was only active for a short period of time even if the broker was the controller. That's why it was removed, it was not really useful. One can use `KafkaController.isActive` to know if a broker is the controller. The question about creating topics seems to be orthogonal.
